.github/workflows/sanity-tests.yml

```
@@ -0,0 +1,43 @@
+name: Sanity tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  # Run once per week (Monday at 04:00 UTC)
+  schedule:
+    - cron: '0 4 * * 1'
+
+jobs:
+  build:
+    name: '${{ matrix.name }}'
+    runs-on: ubuntu-latest
+    strategy:
```

```suggestion
    strategy:
      fail-fast: false
```
Then failing tests for one version (for example because `ansible-galaxy collection install` doesn't work) doesn't cancel all  sanity tests.

---

done